txpipe/twopoint_fourier.py

```diff
@@ -289,8 +270,144 @@ def load_maps(self):
+            for systmap in systmaps_path.iterdir():
+                systmap_file = str(systmap)
+                # check if it is an "_e1" map or a scalar map
+                if ('_e1' in  systmap.name or '_e2' in  systmap.name):
+                    if ('_e1' in  systmap.name):
+                        s_maps_e1_names.append(systmap_file)
+                else:
+                    s_maps_scalar_names.append(systmap_file)
+          
+            # load scalar maps
+            n_systmaps = 0
+            n_systmaps_scalar = 0
+            s_maps_1 = []
+            s_maps_2 = []
+            for file in s_maps_scalar_names:
+                systmap = pathlib.Path(file)
+                try:
+                    if systmap.is_file():
+                        if pathlib.Path(systmap).suffix != ".fits":
+                            print('Warning: Problem reading systematics map file', systmap, 'Not a HEALPix .fits file.')
+                            warnings.warn("Systematics map file must be a HEALPix .fits file.")
+                            print('Ignoring', systmap)
+                        else:
+                            systmap_file = str(systmap)
+                            self.config[f'weaklensing_scalar_deproject_{n_systmaps}'] = systmap_file # for provenance
+                            print('Reading weaklensing systematics map file:', systmap_file)
+                            syst_map = healpy.read_map(systmap_file,verbose=False)
+                            # normalize map for Namaster
+                            # set pixel values to value/mean - 1
+                            syst_map_mask = syst_map != healpy.UNSEEN
+                            mean = np.mean(syst_map[syst_map_mask]) # gives mean of all pixels with mask applied
+                            if mean != 0:
+                                print('Syst map: mean value = ', mean)
+                                syst_map[~syst_map_mask] = 0 # sets unmasked pixels to zero 
+                                syst_map = syst_map / mean - 1
+#                                 print('Syst map', systmap_file, 'normalized value at ra,dec = 55,-30: ', syst_map[ipix])
+
+                            s_maps_1.append(syst_map)
+                            s_maps_2.append(syst_map)
+                            n_systmaps_scalar += 1
+                except:
+                    print('Warning: Problem with systematics map file',systmap)
+                    print('Ignoring', systmap)
+                    
+            print('Number of scalar systematics maps read: ', n_systmaps_scalar)
+                    
+            # load _e1, _e2 maps
+            n_systmaps = 0
+            n_systmaps_e1 = 0
+            for file in s_maps_e1_names:
+                systmap = pathlib.Path(file)
+                try:
+                    if systmap.is_file():
+                        if pathlib.Path(systmap).suffix != ".fits":
```

Comment:
same comment as above for the clustering part.

Reply:
ok